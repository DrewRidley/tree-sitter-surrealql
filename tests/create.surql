-- Create a new record with a text id
CREATE person SET
    name = 'Tobie',
    company = 'SurrealDB',
    skills = ['Rust', 'Go', 'JavaScript'];


-- Create a new record with a numeric id
CREATE person:100 CONTENT {
name: 'Tobie',
company: 'SurrealDB',
skills: ['Rust', 'Go', 'JavaScript'],
};


-- Note: meta::tb(id) returns just the table name portion of a record ID
CREATE townsperson, cat, dog SET
    created_at = time::now(),
    name = "Just a " + meta::tb(id);


-- -- Creates three townperson records with a random ID
-- CREATE |townsperson:3|;



-- CREATE dog, |cat:2|, |townsperson:1..3| SET
--     created_at = time::now(),
--     name = "Just a " + meta::tb(id);



-- -- Returns just a single record
-- CREATE ONLY person:tobie SET
--     name = 'Tobie',
--     company = 'SurrealDB',
--     skills = ['Rust', 'Go', 'JavaScript'];


-- -- Create 10000 records but don't show any of them
-- CREATE |person:10000| SET age = 46, username = "john-smith" RETURN NONE;


-- -- Will always return NONE
-- CREATE person SET age = 46, username = "john-smith" RETURN BEFORE;

-- -- Return the record after creation
-- CREATE person SET age = 46, username = "john-smith" RETURN AFTER;


-- CREATE person
--     SET age = 46,
--     username = "john-smith",
--     interests = ['skiing', 'music']
-- RETURN
--     age,
--     interests,
--     age + 1 AS age_next_year;

-- CREATE |person:5|
--     SET age = 20
-- RETURN VALUE age;

-- -- Query attempting to create half a million `person` records
-- CREATE |person:500000| SET age = 46, username = "john-smith" TIMEOUT 500ms;

-- CREATE person:26, CREATE person:27 PARALLEL;
