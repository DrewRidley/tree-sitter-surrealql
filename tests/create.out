(source_file [0, 0] - [67, 0]
  (comment [0, 0] - [0, 37])
  (statement [1, 0] - [4, 42]
    (create_statement [1, 0] - [4, 41]
      (identifier [1, 7] - [1, 13]
        (raw_identifier [1, 7] - [1, 13]))
      (set_clause [1, 14] - [4, 41]
        (field_assignment [2, 4] - [2, 18]
          (field_reference [2, 4] - [2, 8]
            (identifier [2, 4] - [2, 8]
              (raw_identifier [2, 4] - [2, 8])))
          (expression [2, 11] - [2, 18]
            (value_expr [2, 11] - [2, 18]
              (string [2, 11] - [2, 18]))))
        (field_assignment [3, 4] - [3, 25]
          (field_reference [3, 4] - [3, 11]
            (identifier [3, 4] - [3, 11]
              (raw_identifier [3, 4] - [3, 11])))
          (expression [3, 14] - [3, 25]
            (value_expr [3, 14] - [3, 25]
              (string [3, 14] - [3, 25]))))
        (field_assignment [4, 4] - [4, 41]
          (field_reference [4, 4] - [4, 10]
            (identifier [4, 4] - [4, 10]
              (raw_identifier [4, 4] - [4, 10])))
          (expression [4, 13] - [4, 41]
            (value_expr [4, 13] - [4, 41]
              (array [4, 13] - [4, 41]
                (expression [4, 14] - [4, 20]
                  (value_expr [4, 14] - [4, 20]
                    (string [4, 14] - [4, 20])))
                (expression [4, 22] - [4, 26]
                  (value_expr [4, 22] - [4, 26]
                    (string [4, 22] - [4, 26])))
                (expression [4, 28] - [4, 40]
                  (value_expr [4, 28] - [4, 40]
                    (string [4, 28] - [4, 40]))))))))))
  (comment [7, 0] - [7, 40])
  (statement [8, 0] - [12, 2]
    (create_statement [8, 0] - [12, 1]
      (record_id [8, 7] - [8, 17]
        (raw_identifier [8, 7] - [8, 13])
        (number [8, 14] - [8, 17]))
      (content_clause [8, 18] - [12, 1]
        (object [8, 26] - [12, 1]
          (object_property [9, 0] - [9, 13]
            (object_key [9, 0] - [9, 4]
              (raw_identifier [9, 0] - [9, 4]))
            (expression [9, 6] - [9, 13]
              (value_expr [9, 6] - [9, 13]
                (string [9, 6] - [9, 13]))))
          (object_property [10, 0] - [10, 20]
            (object_key [10, 0] - [10, 7]
              (raw_identifier [10, 0] - [10, 7]))
            (expression [10, 9] - [10, 20]
              (value_expr [10, 9] - [10, 20]
                (string [10, 9] - [10, 20]))))
          (object_property [11, 0] - [11, 36]
            (object_key [11, 0] - [11, 6]
              (raw_identifier [11, 0] - [11, 6]))
            (expression [11, 8] - [11, 36]
              (value_expr [11, 8] - [11, 36]
                (array [11, 8] - [11, 36]
                  (expression [11, 9] - [11, 15]
                    (value_expr [11, 9] - [11, 15]
                      (string [11, 9] - [11, 15])))
                  (expression [11, 17] - [11, 21]
                    (value_expr [11, 17] - [11, 21]
                      (string [11, 17] - [11, 21])))
                  (expression [11, 23] - [11, 35]
                    (value_expr [11, 23] - [11, 35]
                      (string [11, 23] - [11, 35])))))))
          (ERROR [11, 36] - [11, 37])))))
  (comment [15, 0] - [15, 72])
  (statement [16, 0] - [17, 26]
    (create_statement [16, 0] - [17, 26]
      (identifier [16, 7] - [16, 18]
        (raw_identifier [16, 7] - [16, 18]))
      (identifier [16, 20] - [16, 23]
        (raw_identifier [16, 20] - [16, 23]))
      (identifier [16, 25] - [16, 28]
        (raw_identifier [16, 25] - [16, 28]))
      (set_clause [16, 29] - [17, 26]
        (field_assignment [17, 4] - [17, 26]
          (field_reference [17, 4] - [17, 14]
            (identifier [17, 4] - [17, 14]
              (raw_identifier [17, 4] - [17, 14])))
          (expression [17, 17] - [17, 26]
            (value_expr [17, 17] - [17, 26]
              (namespace_call [17, 17] - [17, 26]
                (raw_identifier [17, 17] - [17, 21])
                (raw_identifier [17, 23] - [17, 26]))))))))
  (statement [17, 26] - [18, 36]
    (expression [17, 26] - [18, 35]
      (value_expr [17, 26] - [18, 35]
        (ERROR [17, 27] - [18, 10]
          (ERROR [18, 4] - [18, 6])
          (ERROR [18, 7] - [18, 8]))
        (expression [18, 11] - [18, 35]
          (additive_expr [18, 11] - [18, 35]
            (expression [18, 11] - [18, 20]
              (value_expr [18, 11] - [18, 20]
                (string [18, 11] - [18, 20])))
            (expression [18, 23] - [18, 35]
              (value_expr [18, 23] - [18, 35]
                (function_call [18, 23] - [18, 35]
                  (namespace_call [18, 23] - [18, 31]
                    (raw_identifier [18, 23] - [18, 27])
                    (raw_identifier [18, 29] - [18, 31]))
                  (argument_list [18, 32] - [18, 34]
                    (expression [18, 32] - [18, 34]
                      (value_expr [18, 32] - [18, 34]
                        (field_reference [18, 32] - [18, 34]
                          (identifier [18, 32] - [18, 34]
                            (raw_identifier [18, 32] - [18, 34]))))))))))))))
  (comment [21, 0] - [21, 55])
  (comment [22, 0] - [22, 26])
  (comment [26, 0] - [26, 46])
  (comment [27, 0] - [27, 32])
  (comment [28, 0] - [28, 39])
  (comment [32, 0] - [32, 34])
  (comment [33, 0] - [33, 31])
  (comment [34, 0] - [34, 22])
  (comment [35, 0] - [35, 29])
  (comment [36, 0] - [36, 45])
  (comment [39, 0] - [39, 53])
  (comment [40, 0] - [40, 75])
  (comment [43, 0] - [43, 29])
  (comment [44, 0] - [44, 69])
  (comment [46, 0] - [46, 38])
  (comment [47, 0] - [47, 68])
  (comment [50, 0] - [50, 16])
  (comment [51, 0] - [51, 20])
  (comment [52, 0] - [52, 31])
  (comment [53, 0] - [53, 38])
  (comment [54, 0] - [54, 9])
  (comment [55, 0] - [55, 11])
  (comment [56, 0] - [56, 17])
  (comment [57, 0] - [57, 32])
  (comment [59, 0] - [59, 20])
  (comment [60, 0] - [60, 19])
  (comment [61, 0] - [61, 20])
  (comment [63, 0] - [63, 64])
  (comment [64, 0] - [64, 78])
  (comment [66, 0] - [66, 47]))
tests/create.surql	   0.13 ms	 12738 bytes/ms	(ERROR [11, 36] - [11, 37])
